bitrepository-core/src/main/java/org/bitrepository/common/utils/Base16Utils.java

```diff
@@ -40,35 +42,22 @@ public static String decodeBase16(byte[] data) {
         if (data == null) {
             return null;
         }
-
-        StringBuilder sb = new StringBuilder(data.length * 2);
-        for (byte datum : data) {
-            int v = datum & 0xff;
-            if (v < 16) {
-                sb.append('0');
-            }
-            sb.append(Integer.toHexString(v));
-        }
-        return sb.toString();
+        return Hex.encodeHexString(data);
```

Comment:
Similarly to below one may consider adding a comment along the lines of `TODO Java 17 has HexFormat.of().formatHex(bytes); - consider using instead`.

Reply:
I actually did write this, but somehow I deleted it again. Will add. :)